lib/rules/no-await-sync-events.ts

```diff
@@ -76,6 +76,7 @@ export default createTestingLibraryRule<Options, MessageIds>({
 						property.id.name === 'delay' &&
 						isLiteral(property.init) &&
 						property.init.value &&
+						// @ts-expect-error -- TODO: fix me
 						property.init.value > 0
```

Comment:
Previously, we had `suppressImplicitAnyIndexErrors: true` in `tsconfig.json` which allowed such code. But in TypeScript v5, it complains about this configuration so I removed it.